### ISSUE TITLE
Platforms default value for WATERLOGGED block property is true

### ISSUE THREAD

**Issue by ChiefArug** [SlimeKnights/TinkersConstruct#5225]
### Minecraft Version

1.19.2

### Forge Version

43.3.2

### Mantle Version

1.10.24

### Tinkers' Construct Version

3.8.1.16

### Describe your issue

The default state for platforms is to be waterlogged, so when used in commands like /fill you have to specify that you don't want the block to be waterlogged, which is not how all other blocks work.

How to reproduce:
1. run `/setblock ~ ~2 ~ tconstruct:iron_platform`
2. get wet

This is caused by the default value of boolean block state properties (like WATERLOGGED) being true. This is fixed easily by registering the default state in the block constructor and setting waterlogged to false there.

### Other mods

I trust that none of these mods are causing it as they are other well known, authored by me, or do not touch block properties.
```
rhino, kubejs, jei, ae2, darmstadtium, thermal, kubejs_thermal, 
waystones, xaeroworldmap, shutupexperimentalsettings, controlling, 
modernfix, hardcorequesting, mixinextras, cofh_core, pneumaticcraft, 
creeperconfetti, starlight, cyanide, balm, betterbeaconbeam, 
cloth_config, forge, watchedpot, minecraft, w2w2, tconstruct, systeams, 
thermal_innovation, mousetweaks, markcraft, botania, resourcefulconfig, 
kindled, spark, curios, mantle, patchouli, xaerominimap, 
thermal_cultivation, thermal_foundation, thermal_expansion, badbeacon, 
thermaloot, kubejs_botania, architectury, appleskin, creeperoverhaul, 
ferritecore, probejs, morethan50, smartcompletion, geckolib3, 
thermal_dynamics
```

### Tried reproducing with just Tinkers?

No

### Performance Enchancers

None of the above

### Searched for known issues?

Checked pinned issues, Searched open issues, Searched closed issues, Checked the FAQ

**Comment by KnightMiner** [comment]
Fixed in 3.8.2.32 for 1.19.2. Will leave open until its fixed in 1.18.

**Comment by KnightMiner** [comment]
Fixed in 3.7.2.167 for 1.18.2
